controllers/component_build_controller_test.go

```diff
@@ -193,4 +187,65 @@ var _ = Describe("Component initial build controller", func() {
 			deleteComponent(resourceKey)
 		})
 	})
+
+	Context("Resolve the correct build bundle during the component's creation", func() {
+		It("should use the build bundle specified if a configmap is set in the current namespace", func() {
+			buildBundle := "quay.io/some-repo/some-bundle:0.0.1"
+
+			componentKey := types.NamespacedName{Name: HASCompName, Namespace: HASAppNamespace}
+			configMapKey := types.NamespacedName{Name: prepare.BuildBundleConfigMapName, Namespace: HASAppNamespace}
+
+			createConfigMap(configMapKey, map[string]string{
+				prepare.BuildBundleConfigMapKey: buildBundle,
+			})
+			createComponent(componentKey)
+			setComponentDevfileModel(componentKey)
+
+			ensureOneInitialPipelineRunCreated(componentKey)
+			pipelineRuns := listComponentInitialPipelineRuns(componentKey)
+
+			Expect(pipelineRuns.Items[0].Spec.PipelineRef.Bundle).To(Equal(buildBundle))
+
+			deleteComponent(componentKey)
+			deleteComponentInitialPipelineRuns(componentKey)
+			deleteConfigMap(configMapKey)
+		})
+
+		It("should use the build bundle specified if a configmap is set in the default bundle namespace", func() {
+			buildBundle := "quay.io/some-repo/some-bundle:0.0.2"
+
+			componentKey := types.NamespacedName{Name: HASCompName, Namespace: HASAppNamespace}
+			configMapKey := types.NamespacedName{Name: prepare.BuildBundleConfigMapName, Namespace: prepare.BuildBundleDefaultNamespace}
+
+			createNamespace(prepare.BuildBundleDefaultNamespace)
```

Comment:
The test works as it is, but I'd like to delete this namespace to avoid leaving unused stuff behind. When I try to delete it, though, the test times out, probably because the deletion is not working. 

Not sure what I'm doing wrong:
```
func deleteNamespace(name string) {
	resourceKey := types.NamespacedName{Name: name}
	// Delete
	Eventually(func() error {
		f := &corev1.Namespace{}
		k8sClient.Get(ctx, resourceKey, f)
		return k8sClient.Delete(ctx, f)
	}, timeout, interval).Should(Succeed())

	// Wait for delete to finish
	Eventually(func() error {
		f := &corev1.Namespace{}
		return k8sClient.Get(ctx, resourceKey, f)
	}, timeout, interval).ShouldNot(Succeed())
}
```

Reply:
one thing that can delay namespace deletion is that pod termination within the namespace can get bogged down for any number of reasons; I've seen it happen if Pods mount certain type of volumes for example.

I see a couple of options @brunoapimentel : 
1)  override the delete grace period per the explanation of that field in the `DeleteOptions` i.e.  https://github.com/kubernetes/apimachinery/blob/430b920312ca0fa10eca95967764ff08f34083a3/pkg/apis/meta/v1/types.go#L470-L475 and then use `Delete(ctx context.Context, obj Object, opts ...DeleteOption) error` to get it to process immediately, and let k8s garbage collection proceed in the background
2) if you are really curious about what is delaying things, don't return on the delete, but call it in the background, and the create a poller that inspects Events on the namesapce before you exit;  admittedly a much more involved endeavor :-) .... but if you are curious and need help crafting something like that, let me know

my guess is try 1) first

Reply:
So, setting the grace period to 0 did not work (I remember I had tried it before...)

Reply:
I also tried the force delete instructions from [here](https://stackoverflow.com/questions/55853312/how-to-force-delete-a-kubernetes-namespace#55857926), but no luck.

Reply:
hmmm ... then my 2. is the only think I can think of to suggest to you @brunoapimentel

Reply:
At this point, I guess it's better to merge this PR as it is, and, since the issue is not affecting the tests, we can fix it in a following PR.